src/packages/debugging.md

```diff
@@ -0,0 +1,59 @@
+# Debugging packages
+
+Default CheriBSD package repositories are built with optimisation levels defined
+in CheriBSD ports or upstream third-party software projects.
+Most ports use the `-O2` optimisation level but some switch to `-O3` for better
+performance or `-O1` due to compiler issues preventing them from using
+a higher optimisation level.
+Unfortunately, the FreeBSD ports build system does not allow to build ports with
+optimisations and debug symbols at the same time.
+
+In order to allow users to use optimised and debug CheriABI packages, CheriBSD
+provides two package repositories with them: the default repository with
+optimisations and without debug symbols called `CheriBSD`, and the debug
+repository without optimisations and with debug symbols called `CheriBSD-debug`.
+Note that we do not provide debug packages for the hybrid and benchmark ABIs.
+
+You can enable the debug repository by first creating a directory for custom
+package repository configuration files:
+```
+mkdir -p /usr/local/etc/pkg/repos
+```
+and then creating a configuration file that enables `CheriBSD-debug`:
+```
+echo "CheriBSD-debug: { enabled: yes }" >/usr/local/etc/pkg/repos/CheriBSD-debug.conf
+```
+
+You can reinstall all previously installed default packages to replace them with
+debug packages:
+
+```
+pkg64c upgrade -fr CheriBSD-debug
+```
+
+Alternatively, you can selectively reinstall a package:
+
+```
+pkg64c upgrade -r CheriBSD-debug <pkg-name>
+```
+
+However, keep in mind that in such case only the package you list is
+reinstalled, leaving its dependencies intact.
+
+Once you enable the `CheriBSD-debug` repository, remember to always specify
+which package repository you want to use with the `-r` command flag when
+executing a `pkg64c` command, e.g.
+`pkg64c install -r CheriBSD <pkg-name>` or
+`pkg64c install -r CheriBSD-debug <pkg-name>`.
+Otherwise, it can happen that `CheriBSD-debug` is used whilst you want to use
+the default package repository.
```

Comment:
Is this true? I though `CheriBSD` has precedence?

Reply:
Also, do we want to give an example of disabling `CheriBSD` entirely?

Reply:
> Is this true? I though `CheriBSD` has precedence?

I remember I was getting unexpected results when working with two package repositories enabled at the same time, even if the CheriBSD-debug repository has the priority 1, which makes it less important than the CheriBSD repository with the priority 0.

However, I wasn't able to reproduce this issue now. I've rephrased this paragraph to assume that the CheriBSD repository has precedence here, and I can get back to documenting a potential issue once I can reproduce it.

Reply:
> Also, do we want to give an example of disabling `CheriBSD` entirely?

That's a good idea. I've added a paragraph on that.